graph2tac/tfgnn/q_checkpoint_manager.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.lib.io import file_io
+
+def _delete_file_if_exists(filespec):
+    """Deletes files matching `filespec`."""
+    for pathname in file_io.get_matching_files(filespec):
+        try:
+            file_io.delete_file(pathname)
+        except errors.NotFoundError:
+            logging.warning(
+                "Hit NotFoundError when deleting '%s', possibly because another "
+                "process/thread is also deleting/moving the same file", pathname)
+
+class QCheckpointManager(tf.train.CheckpointManager):
+    def __init__(self, *args, qsaving = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        assert qsaving > 1
```

Comment:
This produces a `TypeError` when qsaving is None. Moreover, asserts should never trigger, so when checking user-provided input we should do proper validation + exception raising if necessary
```suggestion
        if qsaving is not None and qsaving <= 1:
            raise ValueError(f'qsaving parameter should be greater than 1, but got {qsaving}')
```

Reply:
oh, you are right, good catch